synapseclient/credentials/cached_sessions.py

```diff
@@ -0,0 +1,79 @@
+import keyring
+import os
+import json
+import warnings
+from keyring.errors import PasswordDeleteError
+from keyring.backends.fail import Keyring as FailKeyring
+
+
+SYNAPSE_CACHED_SESSION_APLICATION_NAME = "SYNAPSE.ORG_CLIENT"
+SESSION_CAHCE_FILEPATH = os.path.expanduser("~/.synapseSession")
+
+# In the case where no key store backend is available (most likely in Linux), a fail.Keyring is returned which will throw errors when any of its functions are called
+# However, the errors thrown are of the generic type RuntimeError (not a subclass of RuntimeError).
+# It didn't feel safe to try/except and ignore RuntimeError since there are very many other RuntimeErrors that could occur.
+# Instead, we use this boolean to keep track of whether we should call the keyring methods
+_keyring_is_available = not isinstance(keyring.get_keyring(), FailKeyring)
+
+
+def get_api_key(username):
+    """
+    Retrieves the user's API key
+    :param str username:
+    :return: API key for the specified username
+    :rtype: str
+    """
+    if _keyring_is_available and username is not None:
+            return keyring.get_password(SYNAPSE_CACHED_SESSION_APLICATION_NAME, username)
+    return None
+
+
+def remove_api_key(username):
+    if _keyring_is_available:
+        try:
+            keyring.delete_password(SYNAPSE_CACHED_SESSION_APLICATION_NAME, username)
+        except PasswordDeleteError:
```

Comment:
please check to see if you can find a more concrete error or message for password not found.